packages/dialog/src/component.vue

```diff
@@ -133,9 +131,7 @@
           this.$el.removeEventListener('scroll', this.updatePopper);
           if (!this.closed) this.$emit('close');
           if (this.destroyOnClose) {
-            this.$nextTick(() => {
-              this.key++;
-            });
+            this.rendered = false;
```

Comment:
maybe u can move line `133~135` to `afterLeave` method，then the default slot will destroy after the animation ends